src/shutdown/v1/BalancerShutdownStrategy.sol

```diff
@@ -0,0 +1,201 @@
+         .>%%%%%%%%%%%%%%%%%%>>,%%%%%%;,.
+       .>>>>%%%%%%%%%%%%%>>,%%%%%%%%%%%%,>>%%,.
+     .>>%>>>>%%%%%%%%%>>,%%%%%%%%%%%%%%%%%,>>%%%%%,.
+   .>>%%%%%>>%%%%>>,%%>>%%%%%%%%%%%%%%%%%%%%,>>%%%%%%%,
+  .>>%%%%%%%%%%>>,%%%%%%>>%%%%%%%%%%%%%%%%%%,>>%%%%%%%%%%.
+  .>>%%%%%%%%%%>>,>>>>%%%%%%%%%%'..`%%%%%%%%,;>>%%%%%%%%%>%%.
+.>>%%%>>>%%%%%>,%%%%%%%%%%%%%%.%%%,`%%%%%%,;>>%%%%%%%%>>>%%%%.
+>>%%>%>>>%>%%%>,%%%%%>>%%%%%%%%%%%%%`%%%%%%,>%%%%%%%>>>>%%%%%%%.
+>>%>>>%%>>>%%%%>,%>>>%%%%%%%%%%%%%%%%`%%%%%%%%%%%%%%%%%%%%%%%%%%.
+>>%%%%%%%%%%%%%%,>%%%%%%%%%%%%%%%%%%%'%%%,>>%%%%%%%%%%%%%%%%%%%%%.
+>>%%%%%%%%%%%%%%%,>%%%>>>%%%%%%%%%%%%%%%,>>%%%%%%%%>>>>%%%%%%%%%%%.
+>>%%%%%%%%;%;%;%%;,%>>>>%%%%%%%%%%%%%%%,>>>%%%%%%>>;";>>%%%%%%%%%%%%.
+`>%%%%%%%%%;%;;;%;%,>%%%%%%%%%>>%%%%%%%%,>>>%%%%%%%%%%%%%%%%%%%%%%%%%%.
+ >>%%%%%%%%%,;;;;;%%>,%%%%%%%%>>>>%%%%%%%%,>>%%%%%%%%%%%%%%%%%%%%%%%%%%%.
+ `>>%%%%%%%%%,%;;;;%%%>,%%%%%%%%>>>>%%%%%%%%,>%%%%%%'%%%%%%%%%%%%%%%%%%%>>.
+  `>>%%%%%%%%%%>,;;%%%%%>>,%%%%%%%%>>%%%%%%';;;>%%%%%,`%%%%%%%%%%%%%%%>>%%>.
+   >>>%%%%%%%%%%>> %%%%%%%%>>,%%%%>>>%%%%%';;;;;;>>,%%%,`%     `;>%%%%%%>>%%
+   `>>%%%%%%%%%%>> %%%%%%%%%>>>>>>>>;;;;'.;;;;;>>%%'  `%%'          ;>%%%%%>
+    >>%%%%%%%%%>>; %%%%%%%%>>;;;;;;''    ;;;;;>>%%%                   ;>%%%%
+    `>>%%%%%%%>>>, %%%%%%%%%>>;;'        ;;;;>>%%%'                    ;>%%%
+     >>%%%%%%>>>':.%%%%%%%%%%>>;        .;;;>>%%%%                    ;>%%%'
+     `>>%%%%%>>> ::`%%%%%%%%%%>>;.      ;;;>>%%%%'                   ;>%%%'
+      `>>%%%%>>> `:::`%%%%%%%%%%>;.     ;;>>%%%%%                   ;>%%'
+       `>>%%%%>>, `::::`%%%%%%%%%%>,   .;>>%%%%%'                   ;>%'
+        `>>%%%%>>, `:::::`%%%%%%%%%>>. ;;>%%%%%%                    ;>%,
+         `>>%%%%>>, :::::::`>>>%%%%>>> ;;>%%%%%'                     ;>%,
+          `>>%%%%>>,::::::,>>>>>>>>>>' ;;>%%%%%                       ;%%,
+            >>%%%%>>,:::,%%>>>>>>>>'   ;>%%%%%.                        ;%%
+             >>%%%%>>``%%%%%>>>>>'     `>%%%%%%.
+             >>%%%%>> `@@a%%%%%%'     .%%%%%%%%%.
+             `a@@a%@'    `%a@@'       `a@@a%a@@a
+ */
+
+import "src/shutdown/v1/BaseShutdownStrategy.sol";
+
+import {IStrategy} from "src/interfaces/IStrategy.sol";
+import {ILiquidityGauge} from "src/interfaces/ILiquidityGauge.sol";
+import {ReentrancyGuard} from "solady/src/utils/ReentrancyGuard.sol";
+
+/// @notice Strategy contract, supporting Shutdown.
+contract BalancerShutdownStrategy is Ownable2Step, BaseShutdownStrategy, ReentrancyGuard {
+    using SafeERC20 for IERC20;
+
+    /// @notice Address of the strategy.
+    /// @dev It contains most of the storage of the strategy.
+    address public constant STRATEGY = 0x873b031Ea6E4236E44d933Aae5a66AF6d4DA419d;
+
+    /// @notice Address of the BAL token
+    address public constant BAL = 0xba100000625a3754423978a60c9317c58a424e3D;
+
+    /// @notice Address of the BAL minter
+    address public constant BAL_MINTER = 0x239e55F427D44C3cc793f49bFB507ebe76638a2b;
```

Comment:
Would be great to import these addresses from the address book directly (same for `PendleShutdownStrategy` as well)